<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Publications – OurMicrobiome</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark-b53751a350365c71b6c909e95f209ed1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-ac7a74e6b6dd7daa248336fbf4ff6cb4.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">OurMicrobiome</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./OurResearch.html"> 
<span class="menu-text">Our Research</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./people.html"> 
<span class="menu-text">People</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./publications.html" aria-current="page"> 
<span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./training.html"> 
<span class="menu-text">Training</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./news.html"> 
<span class="menu-text">News</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./contact.html"> 
<span class="menu-text">Contact</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#university-of-massachusetts-amherst" id="toc-university-of-massachusetts-amherst" class="nav-link active" data-scroll-target="#university-of-massachusetts-amherst">University of Massachusetts Amherst</a></li>
  <li><a href="#national-center-for-genome-resources-postdoctoral-research" id="toc-national-center-for-genome-resources-postdoctoral-research" class="nav-link" data-scroll-target="#national-center-for-genome-resources-postdoctoral-research">National Center for Genome Resources (Postdoctoral research)</a></li>
  <li><a href="#university-of-oregon-postdoctoral-research" id="toc-university-of-oregon-postdoctoral-research" class="nav-link" data-scroll-target="#university-of-oregon-postdoctoral-research">University of Oregon (Postdoctoral research)</a></li>
  <li><a href="#university-of-georgia-phd-research" id="toc-university-of-georgia-phd-research" class="nav-link" data-scroll-target="#university-of-georgia-phd-research">University of Georgia (PhD research)</a></li>
  <li><a href="#genentech-inc.-between-undergraduate-and-graduate-research" id="toc-genentech-inc.-between-undergraduate-and-graduate-research" class="nav-link" data-scroll-target="#genentech-inc.-between-undergraduate-and-graduate-research">Genentech Inc.&nbsp;(Between undergraduate and graduate research)</a></li>
  <li><a href="#worcester-polytechnic-institute-undergraduate-research" id="toc-worcester-polytechnic-institute-undergraduate-research" class="nav-link" data-scroll-target="#worcester-polytechnic-institute-undergraduate-research">Worcester Polytechnic Institute (Undergraduate research)</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Publications</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p><a href="https://scholar.google.com/citations?user=qSulmx8AAAAJ&amp;hl=en" target="_blank">Google Scholar</a></p>
<section id="university-of-massachusetts-amherst" class="level2">
<h2 class="anchored" data-anchor-id="university-of-massachusetts-amherst">University of Massachusetts Amherst</h2>
<ul>
<li><p>Anthony WE, Allison SD, Broderick CM, Rodriguez LC, Clum A, Cross H, Eloe-Fadrosh E, Evans S, Fairbanks D, Gallery R, Gontijo JB, Jones J, McDermott J, Pett-Ridge J, Record S, Rodrigues JLM, odriguez-Reillo W, Shek KL, Takacs-Vesbach T, Blanchard JL (2024) From soil to sequence: filling the critical gap in genome-resolved metagenomics is essential to the future of soil microbial ecology. Environmental Microbiome 19, 56 (2024). doi.org/10.1186/s40793-024-00599-w</p></li>
<li><p>Graham EB, Camargo AP, Wu R, Neches RY, Nolan M, Paez-Espino D, Kyrpides NC, Jansson JK, McDermott JE, Hofmockel KS; Soil Virosphere Consortium (including Blanchard JL) (2024) A global atlas of soil viruses reveals unexplored biodiversity and potential biogeochemical impacts. Nat Microbiol. 2024 Jul;9(7):1873-1883. doi: 10.1038/s41564-024-01686-x.</p></li>
<li><p>Pakrasi H, Ajo-Franklin C, Donner L, Argueso C, González-Cruz JE, Jones Prather K, Assmann SM, Hungate BA, Schmutz J, Basso B, Juenger T, Segrè D, Blanchard J, Liu X, Shupe MD, Chiao S, Marshall W, Delamere J, Pawlowska T, Taufer M, Fields M, Petch, J, Zeng, X, Fridlind AM (2024) Project Assessment for Biological and Environmental Research: Report from the BER Advisory Committee. https://doi.org/10.2172/2352752</p></li>
<li><p>Fisher M, Mersdorf U, Blanchard JL (2023) Amazing structural diversity of giant virus-like particles in forest soil. <a href="https://www.biorxiv.org/content/10.1101/2023.06.30.546935v1" target="_blank">bioRxiv 2023.06.30.546935</a> <em>This article is a preprint and has not been certified by peer review.</em></p></li>
<li><p>Pavlopoulos GA, Baltoumas FA, Liu S, Selvitopi O, Camargo AP, Nayfach S, Azad A, Roux S, Call L, Ivanova NN, Chen IM, Paez-Espino D, Karatzas E; Novel Metagenome Protein Families Consortium (including Blanchard JL); Iliopoulos I, Konstantinidis K, Tiedje JM, Pett-Ridge J, Baker D, Visel A, Ouzounis CA, Ovchinnikov S, Buluç A, Kyrpides NC (2023). Unraveling the functional dark matter through global metagenomics. Nature. Oct;622(7983):594-602. doi: 10.1038/s41586-023-06583-7.</p></li>
<li><p>Lee B, Neri U, Roux S, Wolf YI, Camargo AP, Krupovic M, RNA Virus Discovery Consortium (including Blanchard JL), Simmonds P, Kyrpides NC, Gophna U, Dolj VV, Koonin EV (2023) A vast world of viroid-like circular RNAs revealed by mining metatranscriptomes. Cell. Jan 21:S0092-8674(22)01582-3. doi: 10.1016/j.cell.2022.12.039.</p></li>
<li><p>Neri U, Wolf YI, Roux S, Camargo AP, Lee B, Kazlauskas D, Chen I-M, Ivanova N, Zeigler Allen L, Paez-Espino D, Bryant DA, Bhaya D, RNA Virus Discovery Consortium (including Blanchard JL), Krupovic M, Dolj VV, Kyrpides NC, Koonin EV, Gophna U (2022) Expansion of the global RNA virome reveals diverse clades of bacteriophages. Cell. 185:21(4023-4037) .e18. doi.org/10.1016/j.cell.2022.08.023</p></li>
<li><p>Fremin BJ, Bhatt AM, Kyrpides NC, Global Phage Small Open Reading Frame (GP-SmORF) Consortium (including Blanchard JL) (2022) Thousands of small, novel genes predicted in global phage genomes. Cell Reports. 39:12. doi.org/10.1016/j.celrep.2022.110984</p></li>
<li><p>Nayfach S, Roux S, Seshadri R, Udwary D, Varghese N, Schulz F, Wu D, Paez-Espino D, Chen IM, Huntemann M, Palaniappan K, Ladau J, Mukherjee S, Reddy TBK, Nielsen T, Kirton E, Faria JP, Edirisinghe JN, Henry CS, Jungbluth SP, Chivian D, Dehal P, Wood-Charlson EM, Arkin AP, Tringe SG, Visel A; IMG/M Data Consortium (including members of our lab Blanchard JL, Alteio LV, Rodriguez-Reillo W)), Woyke T, Mouncey NJ, Ivanova NN, Kyrpides NC, Eloe-Fadrosh EA (2021) A genomic catalogue of Earth’s microbiome Nat Biotechnology 39(4):499-509. doi: 10.1038/s41587-020-0718-6.</p></li>
<li><p>Chowdhury PR, Golas S, Alteio LV, Billings AF, Jeffrey L. Blanchard JL, Melillo JM, DeAngelis KM (2021) The transcriptional response of soil bacteria to long-term warming and short-term seasonal fluctuations in a terrestrial forest. Front Microbiol. Aug 27;12:666558. doi: 10.3389/fmicb.2021.666558.</p></li>
<li><p>Pec GJ van Diepen LTA, Knorr M, Grandy AS, Melillo JM, DeAngelis KM, Blanchard JL, Frey SD (2021). Long-term soil warming selects for a unique fungal community with potential implications for soil carbon dynamics. Ecosphere. https://doi.org/10.1002/ecs2.3460</p></li>
<li><p>LaCroix R, Walpen N, Sander M, Tfaily MM, Blanchard JL, Keiluweit M (2021) Long-term Warming Decreases Redox Capacity of Soil Organic Matter. Environmental Science &amp; Technology Letters. 2021, 8, 1, 92–97 doi: 10.1021/acs.estlett.0c00748</p></li>
<li><p>Haas KN and Blanchard JL (2020) Kineothrix. Bergey’s Manual of Systematics of Archaea and Bacteria. doi: 10.1002/9781118960608.gbm01648</p></li>
<li><p>Andras, J, Rodriguez-Reillo W, Truchon A, Blanchard JL, Pierce E, Ballantine K (2020) Rewilding the small stuff: The effect of ecological restoration on the microbiome of wetland soils. FEMS Microbiol Ecol. 2020 Oct 2;96(10):fiaa144. doi: 10.1093/femsec/fiaa144.</p></li>
<li><p>Alteio L, Schulz F, Seshadri R, Varghese N, Rodriguez-Riello W, Ryan E, Goudeau D, Eichorst S, Malmstrom R, Bowers R, Katz L, Blanchard J, Woyke T (2020) Complementary Metagenomic Approaches Improve Reconstruction of Microbial Diversity in a Forest Soil. mSystems. 2020 Mar 10;5(2). pii: e00768-19. doi: 10.1128/mSystems.00768-19.</p></li>
<li><p>Haas KN, Blanchard JL (2020) Reclassification of the Clostridium clostridioforme and Clostridium sphenoides clades as Enterocloster gen. nov., and Lacrimaspora gen. nov., including reclassification of fifteen taxa. Int J Syst Evol Microbiol. 2020 Jan;70(1):23-34. doi: 10.1099/ijsem.0.003698.</p></li>
<li><p>Cole EF, Zandvakili OR, Blanchard JL, Etemadi F, Hashemi M, Xing B (2019) Investigating soil bacterial community composition and response to hardwood biochar amendment using quantitative PCR. Applied Soil Ecology. Apr;136:80-85. doi.org/10.1016/j.apsoil.2018.12.010</p></li>
<li><p>Schulz F, Alteio L, Goudeau D, Ryan E, Yu G, Malmstrom R, <em>Blanchard J,</em> Woyke T (2018) Hidden diversity of soil giant viruses. Nature Communications. Nov 19;9(1):4881. doi: 10.1038/s41467-018-07335-2.</p></li>
<li><p>Haas KN, Blanchard JL (2017) Kineothrix alysoides, gen. nov., sp. nov., a saccharolytic, butyrate-producer within the family Lachnospiraceae. ISJEM. Feb;67(2):402-410. doi: 10.1099/ijsem.0.001643.</p></li>
<li><p>Pold G, Billings AF, Blanchard JL, Burkhardt DB, Frey SD, Melillo JM, Schnabel J, van Diepen LT, DeAngelis KM (2016) Long-term warming alters carbohydrate degradation potential in temperate forest soils. Appl Environ Microbiol. Oct 27;82(22):6518-6530. doi: 10.1128/AEM.02012-16.</p></li>
<li><p>Tenaillon O, Barrick JE, Ribeck N, Deatherage DE, Blanchard JL, Dasgupta A, Wu GC, Wielgoss S, Cruveiller S, Médigue C, Schneider D, Lenski RE (2016) Tempo and mode of genome evolution in a 50,000-generation experiment. Nature. 536(7615):165-170. doi: 10.1038/nature18959.</p></li>
<li><p>Petit E, Coppi MV, Hayes JC, Tolonen A, Warnick T, Latouf WG, Amisano D, Biddle A, Mukherjee S, Ivanova, N, Lykidis A, Land M, Hauser L, Kripides N, Henrissat B, Lau J, Schnell D, Church G, Leschine SB, Blanchard JL (2015) Genome and transcriptome of Clostridium phytofermentans, catalyst for the direct conversion of plant feedstocks to fuels. PLoS ONE. 10(6): e0118285. doi: 10.1371/journal.pone.0118285</p></li>
<li><p>DeAngelis KM, Pold G, Topcuoglu BD, van Diepen LTA, Varney R, Blanchard JL, Melillo J, Frey SD (2015) Long-term forest soil warming alters microbial communities in temperate forest soils. Frontiers in Microbiology. 6:104. doi: 10.3389/fmicb.2015.00104</p></li>
<li><p>Biddle AS, Leschine S, Huntemann M, Han J, Chen A, Kyrpides N, Markowitz V, Palaniappan K, Ivanova N, Mikhailova N, Ovchinnikova G, Schaumberg A, Pati A, Stamatis D, Reddy T, Lobos E, Goodwin L, Nordberg HP, Cantor MN, Hua SX, Woyke T, Blanchard JL (2014). The complete genome sequence of Clostridium indolis. Standards in Genome Sciences. Mar 18;9(3):1089-104</p></li>
<li><p>Sun Z, Blanchard JL (2014) Strong genome-wide selection early in the evolution of Prochlorococcus resulted in a reduced genome through the loss of a large number of small effect genes. PLoS ONE. Mar 3;9(3):e88837. doi: 10.1371/journal.pone.0088837</p></li>
<li><p>Mukherjee S, Thompson LK, Godin S, Schackwitz W, Lipzen A, Martin J, Blanchard JL (2014) Population level analysis of evolved mutations underlying improvements in plant cellulose and hemicellulose fermentation by Clostridium phytofermentans. PLoS One. Jan 22;9(1):e86731. doi: 10.1371/journal.pone.0086731</p></li>
<li><p>Biddle AS, Black SJ, Blanchard JL (2013) An in vitro model of the horse gut microbiome enables identification of lactate-utilizing bacteria that differentially respond to starch induction. PLoS ONE 8(10): e77599. doi:10.1371/journal.pone.0077599</p></li>
<li><p>Abdul-Rahman F, Petit E, Blanchard JL (2013) The Distribution of Polyhedral Bacterial Microcompartments Suggests Frequent Horizontal Transfer and Operon Reassembly. J Phylogen Evolution Biol. 1:e118</p></li>
<li><p>Biddle AS, Stewart L, Blanchard JL, Leschine SB (2013) Untangling the Genetic Basis of Fibrolytic Specialization by Lachnospiraceae and Ruminococcaceae in Diverse Gut Communities. Diversity 5(3), 627-640</p></li>
<li><p>Andrew C. Tolonen, Elsa Petit, Jeffrey L. Blanchard, Tom Warnick, Susan B. Leschine (2013). “Technologies to Study Plant Biomass Fermentation using the Model Bacterium Clostridium phytofermentans.” Chapt 7 (pp 114-139) in “Biological Conversion of Biomass for Fuels and Chemicals: Explorations from Natural Utilization Systems”. Edited by Jianzhong Sun, Shi-You Ding and Joy Peterson. The Royal Society of Chemistry.</p></li>
<li><p>Petit E, Latouf W, Coppi M, Currie D, Rhomasko I, Desphande S, Haas K, Alvelo-Maurosa JG, Wardman C, Schnell D, Leschine L, Blanchard J (2013) Involvement of a bacterial microcompartment in the metabolism of fucose and rhamnose by Clostridium phytofermentans. PLoS One. 8(1):e54337</p></li>
<li><p>Damery D, Hayes J, Blanchard JL (2010). The potential for production of ethanol from woody biomass by the microbe Clostridium phytofermentans. Wood Structures and Properties ’10. Edited by Kúdela and Lagaňa. Arbora Publishers. p21-25</p></li>
<li><p>Hu J, Blanchard JL (2009) Environmental sequence data from the Sargasso Sea reveal that the characteristics of genome reduction in Prochlorococcus are not a harbinger for an escalation in genetic drift. Molecular Biology and Evolution. Jan;26(1):5-13</p></li>
<li><p>Lu S, Becker KA, Hagen MJ, Yan H, Roberts AL, Mathews LA, Schneider SS, Siegelmann HT, Macbeth KJ, Tirrell SM, Blanchard JL, Jerry DJ (2008) Transcriptional Responses to Estrogen and Progesterone in Mammary Gland Identify Networks Regulating p53 Activity. Endocrinology. Oct;149(10):4809-20</p></li>
<li><p>Gorham J, Parekh S, Leschine S (Main authors). Karem K, Fabel J, Blanchard JL (Contributing authors) (2008) Single-step production of cellulosic ethanol. Biofuels Technology. 1(Q4):37-44</p></li>
<li><p>Blanchard JL, Wholey WY, Conlon EM, Pomposiello PJ (2007) Rapid Changes in Gene Expression Dynamics in Response to Superoxide Reveal SoxRS-Dependent and Independent Transcriptional Networks. PLoS ONE. 2007 Nov 14;2(11):e1186</p></li>
<li><p>Richards TA, Dacks JB, Campbell SA, Blanchard JL, Foster PG, McLeod R, Roberts CW. Evolutionary origins of the eukaryotic shikimate pathway: gene fusions, horizontal gene transfer, and endosymbiotic replacements. Eukaryot Cell. 2006 Sep;5(9):1517-31</p></li>
<li><p>Conlon EM, Alpargu G, Blanchard JL (2006) Comparative genomics approaches for identifying genetic regulatory networks. Chance. 19:45-48</p></li>
<li><p>Blanchard JL (2004) Bioinformatics and Systems Biology, rapidly evolving tools for interpreting plant response to global change. Field Crops Research. 90:117-131</p></li>
</ul>
</section>
<section id="national-center-for-genome-resources-postdoctoral-research" class="level2">
<h2 class="anchored" data-anchor-id="national-center-for-genome-resources-postdoctoral-research">National Center for Genome Resources (Postdoctoral research)</h2>
<ul>
<li><p>de Visser AJM, Hermisson J, Wagner, GP, Ancel LW, Bagheri-Chaichian H, Blanchard JL, Chao L, Cheverud JM, Elena SF, Fontana W, Gibson G, Hansen TF, Krakauer D, Lewontin RC, Ofria C, Rice SH, von Dassow G, Wagner A, Whitlock MC (2003) Evolution and detection of genetic robustness. Evolution. 57:1959-1972</p></li>
<li><p>Küffner RM, Gonzales M, Steadman P, Wlodek DK, Jankowitz RJ, Boinoff JR, Montoya AL, Peterson TF, Bulmore DL, Blanchard JL (2002) PathDB. Nucleic Acids Research Database Issue</p></li>
<li><p>Brinkman FSL, Blanchard JL, Greberg H, Wan I, De Koning A, Av-Gay Y, Brunham RC, Fernandez RC, Finlay BB, Otto SP, Ouellette BF, Keeling P, Hancock REW, Rose AM, SJM Jones (2002). Evidence that plant-like genes in Chlamydia species reflect an ancestral relationship between Chlamydiaceae, cyanobacteria, and the chloroplast. Genome Research. 12:1159-67</p></li>
</ul>
</section>
<section id="university-of-oregon-postdoctoral-research" class="level2">
<h2 class="anchored" data-anchor-id="university-of-oregon-postdoctoral-research">University of Oregon (Postdoctoral research)</h2>
<ul>
<li><p>Blanchard JL, Lynch M (2000) Organelle Genes: why do organellar genes end up in the nucleus? Trends in Genetics. 16:315-320</p></li>
<li><p>de Visser AJM, Zeyl CW, Gerrish PJ, Blanchard JL, Lenski RE (1999) Diminishing returns from mutation supply rate in asexual populations. Science. 283:404-406</p></li>
<li><p>Lynch M, Blanchard JL, Houle D, Kibota T, Schultz S, Vassileva L, Willis J (1999) Spontaneous deleterious mutation. Evolution. 53:645663</p></li>
<li><p>Blanchard JL, Hicks JS (1999) The non-photosynthetic plastid in malarial parasites and other apicomplexans is derived from outside the green plastid lineage. Journal of Eukaryotic Microbiology. 46:367-375</p></li>
<li><p>Lynch M, Blanchard JL (1998) Deleterious mutation accumulation in organelle genomes. Genetica. 102/103:29-39</p></li>
</ul>
</section>
<section id="university-of-georgia-phd-research" class="level2">
<h2 class="anchored" data-anchor-id="university-of-georgia-phd-research">University of Georgia (PhD research)</h2>
<ul>
<li><p>Bruns BU, Blanchard JL, Schmidt GW (1998) On the modulation of thylakoid structure and function upon adaptation to prolonged nitrogen-deficiency. Pp. 3081-3086 in Photosynthesis: Mechanisms and Effects (IV), G. Garab (ed), Kluwer Acad. Publ., Dordrecht</p></li>
<li><p>Blanchard JL, Schmidt GW (1996) Mitochondrial DNA migration events in yeast and mammals: integration by a common end-joining mechanism and alternative perspectives on nucleotide substitution patterns. Molecular Biology and Evolution. 13:537-548</p></li>
<li><p>Blanchard JL (1995) The enigmatic chloroplast ndh genes : from chlororespiration to DNA migration. PhD thesis. University of Georgia</p></li>
<li><p>Blanchard JL, Schmidt GW (1995) Pervasive migration of organellar DNA to the nucleus in plants. Journal of Molecular Evolution. 41:397-406</p></li>
</ul>
</section>
<section id="genentech-inc.-between-undergraduate-and-graduate-research" class="level2">
<h2 class="anchored" data-anchor-id="genentech-inc.-between-undergraduate-and-graduate-research">Genentech Inc.&nbsp;(Between undergraduate and graduate research)</h2>
<ul>
<li>Chen AB, Championsmith AA, Blanchard JL, Niepelt BA, Federici MM, Formento J, Sinicropi DV (1992) Quantitation of E. coli protein impurities in recombinant human interferon. Applied Biochemistry and Biotechnology. 36:137-152</li>
</ul>
</section>
<section id="worcester-polytechnic-institute-undergraduate-research" class="level2">
<h2 class="anchored" data-anchor-id="worcester-polytechnic-institute-undergraduate-research">Worcester Polytechnic Institute (Undergraduate research)</h2>
<ul>
<li><p>Weathers PJ, Blanchard JL (1990) Multiple fruiting of the edible mushroom, Coprinus cinereus, from explants of the stipe. Applied Microbiology and Biotechnology. 33:652-656</p></li>
<li><p>Weathers PJ, Cheetham RD, Blanchard JL, Niedzielski J, Crusberg TC (1987) Biotraps for heavy metal removal and recovery from electroplating wastewater. Society for Industrial Micrbiology (SIM) News 37:9-11</p></li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>