<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.555">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>OurMicrobiome - Training and Lab Expectations</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">OurMicrobiome</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./OurResearch.html"> 
<span class="menu-text">Our Research</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./people.html"> 
<span class="menu-text">People</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./publications.html"> 
<span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./training.html" aria-current="page"> 
<span class="menu-text">Training</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./news.html"> 
<span class="menu-text">News</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./contact.html"> 
<span class="menu-text">Contact</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#welcome" id="toc-welcome" class="nav-link active" data-scroll-target="#welcome">Welcome!</a></li>
  <li><a href="#expectations-and-responsibilities" id="toc-expectations-and-responsibilities" class="nav-link" data-scroll-target="#expectations-and-responsibilities">Expectations and Responsibilities</a>
  <ul class="collapse">
  <li><a href="#big-picture" id="toc-big-picture" class="nav-link" data-scroll-target="#big-picture">Big Picture</a></li>
  <li><a href="#small-picture" id="toc-small-picture" class="nav-link" data-scroll-target="#small-picture">Small Picture</a></li>
  <li><a href="#graduate-students" id="toc-graduate-students" class="nav-link" data-scroll-target="#graduate-students">Graduate students</a></li>
  <li><a href="#undergraduate-students" id="toc-undergraduate-students" class="nav-link" data-scroll-target="#undergraduate-students">Undergraduate Students</a></li>
  </ul></li>
  <li><a href="#code-of-conduct" id="toc-code-of-conduct" class="nav-link" data-scroll-target="#code-of-conduct">Code of Conduct</a>
  <ul class="collapse">
  <li><a href="#essential-policies" id="toc-essential-policies" class="nav-link" data-scroll-target="#essential-policies">Essential Policies</a></li>
  <li><a href="#scientific-integrity" id="toc-scientific-integrity" class="nav-link" data-scroll-target="#scientific-integrity">Scientific Integrity</a>
  <ul class="collapse">
  <li><a href="#research-misconduct" id="toc-research-misconduct" class="nav-link" data-scroll-target="#research-misconduct">Research (Mis)conduct</a></li>
  </ul></li>
  <li><a href="#reproducible-research" id="toc-reproducible-research" class="nav-link" data-scroll-target="#reproducible-research">Reproducible Research</a></li>
  <li><a href="#authorship" id="toc-authorship" class="nav-link" data-scroll-target="#authorship">Authorship</a></li>
  <li><a href="#old-projects" id="toc-old-projects" class="nav-link" data-scroll-target="#old-projects">Old projects</a></li>
  </ul></li>
  <li><a href="#general-policies" id="toc-general-policies" class="nav-link" data-scroll-target="#general-policies">General Policies</a>
  <ul class="collapse">
  <li><a href="#hours" id="toc-hours" class="nav-link" data-scroll-target="#hours">Hours</a></li>
  <li><a href="#pi-office-hours" id="toc-pi-office-hours" class="nav-link" data-scroll-target="#pi-office-hours">PI Office Hours</a></li>
  <li><a href="#communication" id="toc-communication" class="nav-link" data-scroll-target="#communication">Communication</a>
  <ul class="collapse">
  <li><a href="#e-mail" id="toc-e-mail" class="nav-link" data-scroll-target="#e-mail">E-mail</a></li>
  <li><a href="#slack" id="toc-slack" class="nav-link" data-scroll-target="#slack">Slack</a></li>
  <li><a href="#google-drive" id="toc-google-drive" class="nav-link" data-scroll-target="#google-drive">Google Drive</a></li>
  <li><a href="#google-calendars" id="toc-google-calendars" class="nav-link" data-scroll-target="#google-calendars">Google Calendars</a></li>
  </ul></li>
  <li><a href="#meetings" id="toc-meetings" class="nav-link" data-scroll-target="#meetings">Meetings</a>
  <ul class="collapse">
  <li><a href="#weekly-lab-meetings" id="toc-weekly-lab-meetings" class="nav-link" data-scroll-target="#weekly-lab-meetings">Weekly Lab Meetings</a></li>
  <li><a href="#deadlines" id="toc-deadlines" class="nav-link" data-scroll-target="#deadlines">Deadlines</a></li>
  <li><a href="#presentations" id="toc-presentations" class="nav-link" data-scroll-target="#presentations">Presentations</a></li>
  <li><a href="#recommendation-letters" id="toc-recommendation-letters" class="nav-link" data-scroll-target="#recommendation-letters">Recommendation Letters</a></li>
  </ul></li>
  <li><a href="#open-science" id="toc-open-science" class="nav-link" data-scroll-target="#open-science">Open Science</a></li>
  <li><a href="#funding" id="toc-funding" class="nav-link" data-scroll-target="#funding">Funding</a></li>
  <li><a href="#acknowledgements" id="toc-acknowledgements" class="nav-link" data-scroll-target="#acknowledgements">Acknowledgements</a></li>
  </ul></li>
  <li><a href="#unity" id="toc-unity" class="nav-link" data-scroll-target="#unity">Unity</a>
  <ul class="collapse">
  <li><a href="#quick-links" id="toc-quick-links" class="nav-link" data-scroll-target="#quick-links">Quick links</a></li>
  <li><a href="#unity-ondemand" id="toc-unity-ondemand" class="nav-link" data-scroll-target="#unity-ondemand">Unity OnDemand</a></li>
  <li><a href="#home-work-and-project-directories" id="toc-home-work-and-project-directories" class="nav-link" data-scroll-target="#home-work-and-project-directories">Home, Work and Project Directories</a></li>
  <li><a href="#using-software-installed-on-unity-and-through-conda" id="toc-using-software-installed-on-unity-and-through-conda" class="nav-link" data-scroll-target="#using-software-installed-on-unity-and-through-conda">Using software installed on Unity and through Conda</a></li>
  <li><a href="#transfering-data-from-unity-to-biologys-network-attached-storage-nas" id="toc-transfering-data-from-unity-to-biologys-network-attached-storage-nas" class="nav-link" data-scroll-target="#transfering-data-from-unity-to-biologys-network-attached-storage-nas">Transfering data from Unity to Biology’s Network Attached Storage (NAS)</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Training and Lab Expectations</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="welcome" class="level1">
<h1>Welcome!</h1>
<p>Welcome to the Blanchard research group. We have put together a lab manual to make expectations and responsibilities clear. If you have ideas about things to add, or what to clarify, let me know. When you join the lab, you’re expected to read this manual and sign a form indicating that you have done so.</p>
</section>
<section id="expectations-and-responsibilities" class="level1">
<h1>Expectations and Responsibilities</h1>
<section id="big-picture" class="level2">
<h2 class="anchored" data-anchor-id="big-picture">Big Picture</h2>
<p>Science is hard. But it’s also fun. In the Blanchard Lab, we want to make sure that everyone experiences a positive, engaging, challenging, and rewarding lab environment. To maintain that environment, we all have to do a few things.</p>
<ul>
<li>Work on what you’re passionate about, work hard at it, and be proud of it. Be so proud of it that you have to suppress bragging (but it’s ok to brag sometimes).</li>
<li>Scientists have to be careful. Don’t rush your work. Think about it. Implement it. Double and triple check it. Incorporate sanity checks. Ask others to look at your code or data if you need help or something looks off. It’s ok to makes mistakes, but mistakes shouldn’t be because of carelessness or rushed work.</li>
<li>If you do make a mistake, you should definitely tell your collaborators (if they have already seen the results, and especially if the paper is being written up, is already submitted, or already accepted). We admit our mistakes, and then we correct them and move on.</li>
<li>We all want to get papers published and do great things. But we do this honestly. It is never ok to plagiarize, tamper with data, make up data, omit data, or fudge results in any way. Science is about finding out the truth, and null results and unexpected results are still important. This can’t be emphasized enough: no academic misconduct!</li>
<li>Support your fellow lab-mates. Help them out if they need help (even if you aren’t on the project), and let them vent when they need to. Science is collaborative, not competitive. Help others, and you can expect others to help you when you need it.</li>
<li>Respect your fellow lab-mates. Respect their strengths and weaknesses, respect their desire for quiet if they need it, and for support and a kind ear when they need that. Respect their culture, their religion, their beliefs, their sexual orientation.</li>
<li>If you’re struggling, tell someone (feel free to tell Jeff!). Your health and happiness come first. The lab looks out for the well-being of all its members. We are here to help. It’s ok to go through hard patches (we all do), but you shouldn’t feel shy about asking for help or just venting.</li>
<li>If there is any tension or hostility in the lab, something has to be done about it immediately. We can’t thrive in an environment we aren’t comfortable in, and disrespect or rudeness will not be tolerated in the lab. If you don’t feel comfortable confronting the person in question, tell me.</li>
<li>If you have a problem with Jeff and are comfortable telling me about it, do! If you aren’t comfortable, then tell the the graduate program director.</li>
<li>Stay up to date on the latest research, by using RSS feeds and/or getting journal table of contents. Also consider following scientists in the field on Twitter.</li>
<li>Remember the lab philosophy: “We like to do good science and have fun. At the same time, but also separately”. Have a life outside of the lab, take care of your mental and physical health, and don’t ever feel bad for taking time off work.</li>
</ul>
</section>
<section id="small-picture" class="level2">
<h2 class="anchored" data-anchor-id="small-picture">Small Picture</h2>
<p>There are a few day-to-day things to keep in mind to keep the lab running smoothly.</p>
<ul>
<li>If you’re sick, stay home and take care of yourself. Because you need it, and also because others don’t need to get sick. If you’re sick, reschedule your meetings and participants for the day (or the next couple of days) as soon as you can.</li>
<li>You aren’t expected to come into lab on weekends and holidays, and you aren’t expected to stay late at night. You are expected to get your work done (whatever time of day you like to do it).</li>
<li>Show up to your meetings, show up to your classes, and show up to lab meetings. You do not have to be in at 9am every day – just show up for your commitments, and work the hours you need to work to get your researh done.</li>
<li>Make sure the door to the lab is locked if no one is inside. Turn off the lights if you’re the last one leaving for the day.</li>
<li>Keep the lab tidy. Eating at your desk (but not in the lab) is fine, but clean up food waste, crumbs, spills. Put lab equipment back where you found it. Keep common areas uncluttered.</li>
<li>Dress code is casual. If you are working in the lab be sure to follow laboratory safety guidelines Principal Investigator All of the above, and I promise to also…</li>
<li>Be available in person and via e-mail on a regular basis, including regular meetings to discuss your research (and anything else you’d like to discuss)</li>
<li>Give my perspective on where the lab is going, where the field is going, and tips about surviving and thriving in academia</li>
<li>Support your career development by introducing you to other researchers in the field, promoting your work at talks, writing recommendation letters for you, and letting you attend conferences as often as finances permit</li>
<li>Help you prepare for the next step of your career, whether it’s an industry job, graduate school, medical school, a post-doc, a faculty job, or a job outside of academia</li>
<li>Give you feedback on a timely basis, including feedback on project ideas, conference posters, talks, manuscripts, figures, grants</li>
<li>Support you (scientifically and financially)</li>
</ul>
</section>
<section id="graduate-students" class="level2">
<h2 class="anchored" data-anchor-id="graduate-students">Graduate students</h2>
<p>All of the above, and you will also be expected to</p>
<ul>
<li>Develop your dissertation research. Your dissertation should have at least 3 substantial chapters that answer a big-picture question that you have. Much of your work has to be done independently, but remember that others in lab are there to help you when you need it and that manuscript publications can include collaborators.</li>
<li>Challenge me when I’m wrong or when your opinion is different, and treat the rest of the lab to your unique expertise.</li>
<li>Help mentor undergraduate students in the lab when they need it – either because they ask, or because I ask you to.</li>
<li>Make sure you meet all departmental deadlines (e.g., for your exams and thesis)</li>
<li>Prioritize time for research. Coursework and TAing are important, but ultimately your research gets you your PhD and prepares you for the next stage of your career.</li>
<li>Apply for grants. It’s in your best interest to get experience writing grants – and if you get them, you’ll be helping out the entire lab as well as yourself (because you’ll free up funds previously allocated to you)</li>
<li>Apply for jobs (academic or otherwise) when you’re ready, but no later than the beginning of your 4th year of graduate school. If you think you’d like to leave academia, that’s completely ok – but you should still treat your disseration seriously, and talk to me about how to best train for a job outside academia.</li>
<li>Present your work at departmental events, at other labs (if invited), and at conferences.</li>
</ul>
</section>
<section id="undergraduate-students" class="level2">
<h2 class="anchored" data-anchor-id="undergraduate-students">Undergraduate Students</h2>
<p>All of the above, and you will also be expected to - Assist other lab members with data collection and analysis (unless you are working on your own independent project under the mentorship of another lab member, in which case you should work on that). - Develop your weekly schedule by talking to your graduate student mentor or your post-doc mentor. You should be coming in every week, and scheduling enough time to get your work done. - If you are earning course credit for research, you must also attend lab meetings when your schedule permits, present at one of these lab meetings, and submit a write-up of your research by the end of the semester.</p>
</section>
</section>
<section id="code-of-conduct" class="level1">
<h1>Code of Conduct</h1>
<section id="essential-policies" class="level2">
<h2 class="anchored" data-anchor-id="essential-policies">Essential Policies</h2>
<p>The lab, and the university, is an environment that must be free of harassment and discrimination. All lab members are expected to abide by the University of Massachusetts Amherst policies on discrimination and harassment, which you can (and must) read about here - https://www.umass.edu/equalopportunity/. The lab is committed to ensuring a safe, friendly, and accepting environment for everybody. We will not tolerate any verbal or physical harassment or discrimination on the basis of gender, gender identity and expression, sexual orientation, disability, physical appearance, body size, race, or religion. We will not tolerate intimidation, stalking, following, unwanted photography or video recording, sustained disruption of talks or other events, inappropriate physical contact, and unwelcome sexual attention. Finally, it should go without saying that lewd language and behavior have no place in the lab, including any lab outings. If you notice someone being harassed, or are harassed yourself, tell Jeff immediately. If Jeff is the cause of your concern, then reach out to the department chair or another trusted faculty member in the department.</p>
</section>
<section id="scientific-integrity" class="level2">
<h2 class="anchored" data-anchor-id="scientific-integrity">Scientific Integrity</h2>
<section id="research-misconduct" class="level3">
<h3 class="anchored" data-anchor-id="research-misconduct">Research (Mis)conduct</h3>
<p>The lab, and University of Massachusetts Amherst, is committed to ensuring research integrity, and we take a hard line on research misconduct. We will not tolerate fabrication, falsification, or plagiarism. Read our university’s policies on the conduct of research carefully - https://www.umass.edu/research/policy/responsible-conduct-research-trustee-policy-t08-10 A big problem is why people feel the need to engage in misconduct in the first place, and that’s a discussion that we can have. If you are feeling pressured to succeed (publish a lot, publish in high impact journals), you should reach out and we can talk about it – but this pressure is something we all face and is never an excuse to fabricate, falsify, or plagiarize. Also, think about the goal of science and why you are here: you’re here to arrive at the truth, to get as close as we can to facts about microbes and their communities. Not only is research misconduct doing you a disservice, it’s also a disservice to the field. And it risks your entire career. It is never right and never worth it. Don’t do it.</p>
</section>
</section>
<section id="reproducible-research" class="level2">
<h2 class="anchored" data-anchor-id="reproducible-research">Reproducible Research</h2>
<p>If you gave someone else your raw data, they should be able to reproduce your results exactly. This is critical, because if they can’t reproduce your results, it suggests that one (or both) of you has made errors in the analysis, and the results can’t be trusted. Reproducible research is an essential part of science, and an expectation for all projects in the lab. For results to be reproducible, the analysis pipeline must be organized and well documented. To meet these goals, you should take extensive notes on each step of your analysis pipeline. This means writing down how you did things every step of the way (and the order that you did things), from any pre-processing of the data, to running models, to statistical tests. It’s also worth mentioning that you should take detailed notes on your experimental design as well. Additionally, your code should also be commented, and commented clearly. We all know what it’s like to sit down, quickly write a bunch of code to run an analysis without taking time to comment it, and then having no idea what we did a few months down the road. Comment your code so that every step is understandable by an outsider. Finally, it is highly encouraged that you use some form of version control (e.g., Git in combination with GitHub) to keep track of what code changes you made and when you made them, as well as sharing code with others. The lab’s GitHub is https://github.com/jeffreyblanchard/ and the lab’s BitBucket is https://bitbucket.org/ourmicrobiome/ Reproducibility is related to replicability, which refers to whether your results can be obtained again with a different data set. That is, if someone ran your study again (with a different group of participants), do they get the same results? If someone ran a conceptually similar study, do they get the same results? Science grows and builds on replicable results – one-off findings don’t mean anything. Our goal is to produce research that is both reproducible and replicable.</p>
</section>
<section id="authorship" class="level2">
<h2 class="anchored" data-anchor-id="authorship">Authorship</h2>
<p>We will follow the APA guidelines with respect to authorship: “Authorship credit should reflect the individual’s contribution to the study. An author is considered anyone involved with initial research design, data collection and analysis, manuscript drafting, and final approval. However, the following do not necessarily qualify for authorship: providing funding or resources, mentorship, or contributing research but not helping with the publication itself. The primary author assumes responsibility for the publication, making sure that the data are accurate, that all deserving authors have been credited, that all authors have given their approval to the final draft; and handles responses to inquiries after the manuscript is published.” At the start of a new project, the student or post-doc taking on the lead role can expect to be first author (talk to Jeff about it if you aren’t sure). Jeff will typically be the last author, unless the project is primarily under the guidance of another PI and Jeff is involved as a secondary PI – then Jeff will be second to last and the main PI will be last. Students and post-docs who help over the course of the project may be added to the author list depending on their contribution, and their placement will be discussed with all parties involved in the paper. If a student or post-doc takes on a project but subsequently hands it off to another student or post-doc, they will most likely lose first-authorship to that student or post-doc, unless co-first-authorship is appropriate. All of these issues will be discussed openly, and you should feel free to bring them up if you are not sure of your authorship status or want to challenge it.</p>
</section>
<section id="old-projects" class="level2">
<h2 class="anchored" data-anchor-id="old-projects">Old projects</h2>
<p>If a student or post-doc collects a dataset but does not completely analyze it or write it after leaving the lab, Jeff will re-assign the project (if appropriate) to another person to expedite publication. In some cases are data sets are so large that there are additional projects that are not directly related to the dissertation that can be spun out. The person who collected the data will still be including as a co-author on publications (unless the entire data set is already published). This policy is here to prevent data (especially expensive data) from remaining unpublished.</p>
</section>
</section>
<section id="general-policies" class="level1">
<h1>General Policies</h1>
<section id="hours" class="level2">
<h2 class="anchored" data-anchor-id="hours">Hours</h2>
<p>Being in lab is a good way of learning from others, helping others, building camaraderie, having fast and easy access to resources (and people) you need, and being relatively free from distractions at home (e.g., your bed or Netflix). That said, hours in academia are more flexible than other jobs – but you should still treat it as a real job (40 hours/week) and show up to the lab. My primary concern is that you get your work done, so if you find that you are more productive at home (lab-mates can be chatty sometimes), feel free to work at home occasionally. If you have no meetings, no participants, and no other obligations that day, it might be a good day to work at home – but you can’t do this all the time, and I expect to see everyone in the lab on a regular basis. The only exception to this is lab managers / research assistants, who must keep more regular hours and be in lab 5 days a week (excluding vacations, doctor appointments, family issues, etc). I expect lab managers / research assistants to be in about 8 hours a day, starting around 9am or 10am and ending around 5pm or 6pm. For graduate students, I understand having to be away for classes and TA-ing, but show up to the lab on a regular basis when you don’t have those obligations. To encourage lab interaction, try to be in most weekdays during ‘peak’ hours (assuming no other obligations) – e.g., between 11am and 4pm. This is not a hard rule, you can work at home occasionally, and I understand other obligations. But keep it in mind.</p>
</section>
<section id="pi-office-hours" class="level2">
<h2 class="anchored" data-anchor-id="pi-office-hours">PI Office Hours</h2>
<p>In addition to weekly meetings (see below), and occasionally dropping by the lab, you can find Jeff in his office. His door is almost always open; if it is, feel free to ask for a chat. He will always say yes, though sometimes he can only spare a couple of minutes. If his door is closed, assume that Jeff is either gone, in a meeting in his office, or does not want to be disturbed – so please send a message (Slack or e-mail) rather than knocking.</p>
</section>
<section id="communication" class="level2">
<h2 class="anchored" data-anchor-id="communication">Communication</h2>
<section id="e-mail" class="level3">
<h3 class="anchored" data-anchor-id="e-mail">E-mail</h3>
<p>I use email as the primary means of communication, but for general lab discussion it is nice to have some level of open views of the discussions over time related to lab meetings, projects and papers. For this we will use Slack. A rule of thumb is if you want to get a hold of me quickly use email or phone or stop in. If you want to send me a summary of our weekly meeting send it on Slack or if you want to communicate with part or all of the lab use Slack.</p>
</section>
<section id="slack" class="level3">
<h3 class="anchored" data-anchor-id="slack">Slack</h3>
<p>Slack will be used as the primary means of lab communication. When posting messages or looking for updates, check the appropriate channel: #general for lab announcements, #lab-meetings for notes or communication related to lab meetings, #papers for sharing links to lab-relevant papers and discussing them, #code-tips for sharing wisdom on code writing or asking (and answering) the coding questions of others, #genomic-methods for sharing wisdom on genome sequencing / analysis or asking (and answering) the questions of others, #stats to ask and answer questions about statistical analyses, and #random for non-work-related chatting that is best kept out of the work-related channels. Try to keep each channel on topic, so that people can subscribe only to the channels that concern them. For messages to one person or a small group, use direct messages. If you have to send attachments (e.g., papers) or send messages that include out-of-lab recipients, put in our GDrive and add a link on Slack. Full-time lab members should install Slack on their computers and/or phones. Part-time lab members should also check Slack regularly. You should of course feel free to ignore Slack on evenings and weekends – and Jeff probably will, too!</p>
</section>
<section id="google-drive" class="level3">
<h3 class="anchored" data-anchor-id="google-drive">Google Drive</h3>
<p>Please post lab related documents to Google Drive using the OurMicrobiome directory rather than including them in email attachments. You can then send a link to the file in your email or on Slack.</p>
</section>
<section id="google-calendars" class="level3">
<h3 class="anchored" data-anchor-id="google-calendars">Google Calendars</h3>
<p>The lab has a lab calendar that include’s Jeff work-related calendar. Ask me to share it if you haven’t received an email yet.</p>
</section>
</section>
<section id="meetings" class="level2">
<h2 class="anchored" data-anchor-id="meetings">Meetings</h2>
<section id="weekly-lab-meetings" class="level3">
<h3 class="anchored" data-anchor-id="weekly-lab-meetings">Weekly Lab Meetings</h3>
<p>Weekly lab meetings (~1 hours each) are meant to be a forum for trainees to present project ideas and/or data to get feedback from the rest of the group. Projects at any level of completion (or even not yet started!) can benefit from being presented. These lab meetings can also be used to talk about methods, statistical analyses, new papers, and career development. For paper discussions, everyone must come to lab meeting having read the paper and prepared with comments and questions to contribute. Some weeks we may explore a particular issue and have people read different papers – in that case, come to lab meeting having read your paper and be prepared to summarize it for the group. Each trainee (RA, students, post-docs) is expected to present at least once every semester. These meetings are informal, and you can do what you wish with your slot – just be prepared to contribute something substantive. Lab members are also expected to attend every meeting (obviously, illnesses, doctor appointments, family issues, etc are a valid reason for missing a meeting). Undergraduate students are encouraged to attend as often as possible (assuming it fits in their course schedule). Occasionally, we may have joint lab meetings with other faculty in the department – these may be combined with our weekly lab meeting or an additional meeting. We will also use lab meetings (or ad-hoc scheduled meetings) to prepare for conference presentations and give people feedback on job talks or other external presentations. Lab meeting agendas and notes will be kept in the #lab-meetings channel on Slack. Individual Meetings At the beginning of each semester, we will set a schedule for weekly meetings. Each full-time lab member (RAs, graduate students, post-docs) will have a one-hour slot set aside to meet with Jeff. If scheduling conflicts arise (e.g., because of travel), we can try to reschedule for another day that week. If there is nothing to discuss, feel free to cancel the meeting or just drop by for a brief chat. Jeff will meet with undergraduate students as a group every week (or according to need); post-docs and graduate students should meet with their undergraduate mentee on a regular basis.</p>
</section>
<section id="deadlines" class="level3">
<h3 class="anchored" data-anchor-id="deadlines">Deadlines</h3>
<p>One way of maintaining sanity in the academic work is to be as organized as possible. This is essential because disorganization will hurt you, it hurts your collaborators and people whose help you need. When it comes to deadlines, tell your collaborators as soon as possible when you know when a deadline is, and make sure they are aware of it the closer it gets. Don’t be afraid to bug them about it (yes, bug Jeff as well). Give Jeff at least one week’s notice to do something with a hard deadline that doesn’t require a lot of time (e.g., reading/commenting on conference abstracts, filling out paperwork, etc). Give Jeff at least two weeks’ notice to do something with a hard deadline that requires a lot of time (e.g., a letter of recommendation). For manuscript revisions and invited paper submissions (which have hard-ish deadlines), give her as much time as you can, because these will require multiple back-and-forths. For manuscript submissions (i.e., no hard deadline), you can still bug Jeff to give you feedback if he hasn’t responded in a week or two – papers are important!</p>
</section>
<section id="presentations" class="level3">
<h3 class="anchored" data-anchor-id="presentations">Presentations</h3>
<p>Learning to present your research is important. Very few people will read your papers carefully (sad, but true) but you can reach a lot of people at conference talks and posters. Also, if you plan on staying in academia, getting a post-doc position and getting a faculty position both significantly depend on your ability to present your data. Even if you want to leave academia, presentations are likely to be an important part of your job. Additionally, every time you present your work, you are representing not just yourself but the entire lab. It is therefore highly encouraged that you seek out opportunities to present your research, whether it is at departmental talk series and events, to other labs (within or outside of Columbia), at conferences, or to the general public. If you are going to give a presentation (a poster or a talk), be prepared to give a practice presentation to the lab at least one week ahead of time (two weeks or more are advisable for conference presentations, and many weeks ahead of time are advisable for job talks, which require much refining). Practice talks will help you feel comfortable with your presentation, and will also allow you to get feedback from the lab and implement those changes well in advance of your real presentation. Templates for posters will be available, and you can use those as much or as little as you’d like. Some general rules for posters should be followed: minimize text as much as possible (if you wrote a paragraph, you’re doing it wrong), make figures and text large and easy to see at a distance, label your axes, and make sure different colors are easily discriminable. Other than that, go with your own style. Jeff is also happy to share slides from talks if you would like to use a similar style. You’ll get a lot of feedback on your talks in any case, but other people’s slides might be helpful to you as you are setting up your talk. As with posters, feel free to go with your own style as long as it is polished and clear.</p>
</section>
<section id="recommendation-letters" class="level3">
<h3 class="anchored" data-anchor-id="recommendation-letters">Recommendation Letters</h3>
<p>Letters of recommendation are extremely important for getting new positions and grants. You can count on Jeff to write you a letter if you have been in the lab at least one year (it’s hard to really know someone if they have only been around for a few months). Exceptions can be made if students or post-docs are applying for fellowships shortly after starting in the lab. If you need a letter, notify Jeff as soon as possible with the deadline (see Deadlines for guidance), your CV, and any relevant instructions for the content of the letter. If the letter is for a grant, also include your specific aims. In some cases (especially if short notice is given), you may also be asked to submit a draft of a letter, which will be modified based on Jeff’s experience with you and anything else that has to be added. This will ensure that the letter contains all the information you need, and that it is submitted on time.</p>
</section>
</section>
<section id="open-science" class="level2">
<h2 class="anchored" data-anchor-id="open-science">Open Science</h2>
<p>We’re all for open science, so lab members are encouraged (well, required) to share their code and data with others, whether they are in the lab or outside of it. Within lab, you can share your code and data whenever you like. But do not share your code or data with the outside world until you think (and Jeff agrees) that the lab has finished working with it. This gives us an opportunity to work with the data to meet our needs (including grant needs!) before releasing it for other people to use. Generally, we will try to make our data and code publicly available within one year of publishing the results (longer if work on the dataset is ongoing). Currently, the best option for sharing smaller datasets might be the Open Science Framework. We will also share our work with the world as soon as we ready, which means preprints! The lab policy is to upload a preprint of a manuscript simultaneously with initial submission to a journal. The preferred preprint servers are bioRxiv. We will also put PDFs of all our papers on the lab website, and you should share PDFs of your paper with whoever asks.</p>
</section>
<section id="funding" class="level2">
<h2 class="anchored" data-anchor-id="funding">Funding</h2>
<p>Funding for the lab currently comes National Science Foundation, Department of Energy and USDA grants. If you need to buy something, or have to charge a grant for something, let Jeff know and he will oversee the process. At some point, you will likely be asked to provide a figure or two for a grant Jeff is writing, and/or provide feedback on the grant. You are entitled to read any grant Jeff has submitted, whether it is ultimately funded or not. Aside from being a good opportunity to learn how grants are written, this will also allow you to see his vision for the lab in the years ahead. Feel free to ask Jeff to see any of the grants.</p>
</section>
<section id="acknowledgements" class="level2">
<h2 class="anchored" data-anchor-id="acknowledgements">Acknowledgements</h2>
<p>Some text on this page was taken directly from an older version of Mariam Aly’s lab manual. Here is a link to the <a href="https://github.com/alylab/labmanual/blob/master/aly-lab-manual.pdf">current version</a>.</p>
</section>
</section>
<section id="unity" class="level1">
<h1>Unity</h1>
<p>Unity is our UMass High Performance Computing Cluster. Here is are some tips for getting started.</p>
<section id="quick-links" class="level2">
<h2 class="anchored" data-anchor-id="quick-links">Quick links</h2>
<ul>
<li><p><a href="https://unity.rc.umass.edu/index.php">Unity Home Page and Login</a> Request an account listing under your PI (e.g.&nbsp;pi_jlb_umass_edu)</p></li>
<li><p><a href="https://docs.unity.rc.umass.edu/">Unity Documentation Site</a> Much useful information here. If you don’t find your answer here, ask on Slack.</p></li>
<li><p><a href="https://docs.unity.rc.umass.edu/contact/community/">Unity User Community on Slack</a> The Unity staff is good at responding to Slack posts (usually within a day)</p></li>
<li><p><a href="https://docs.unity.rc.umass.edu/events/">Unity Onboarding and Bioinformatics workshops</a> These are a nice way to get started and develop your skills</p></li>
<li><p><a href="https://docs.unity.rc.umass.edu/documentation/get-started/quickstart/">Unity Quick Start</a></p></li>
<li><p><a href="https://docs.unity.rc.umass.edu/documentation/tools/r/">Unity R Documentation</a></p></li>
</ul>
</section>
<section id="unity-ondemand" class="level2">
<h2 class="anchored" data-anchor-id="unity-ondemand">Unity OnDemand</h2>
<p>The easiest way to use Unity is to select <code>Open OnDemand</code> after you log in from the menu on the left. On the Dashboard select <code>Interactive Apps</code> to choose your preferred interface. Starting one of these apps is essentially starting a job on Unity. We typically use the Rstudio, Jupyter or VSCode interfaces. You need to specify the resources needed and time. For much of the basic work we do the defaults of 2 CPU and 8 GB Memory are fine. Sometimes if we am running gene expression analysis or something that requires more compute power up these to 8 and 32. If you are running specific software that requires more than 5 minutes or more computer power to run, you should submit a <a href="https://docs.unity.rc.umass.edu/documentation/jobs/sbatch/">batch job using Slurm</a>.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/Unity_OnDemand.png" class="img-fluid figure-img" width="400"></p>
<figcaption>Open OnDemand Dashboard</figcaption>
</figure>
</div>
</section>
<section id="home-work-and-project-directories" class="level2">
<h2 class="anchored" data-anchor-id="home-work-and-project-directories">Home, Work and Project Directories</h2>
<p>Your <code>/home directory</code> is for configuration files and code installations. It has a set quota of 50G. You should do all of your work in our lab work directory (<code>/work/pi_jlb_umass_edu</code>). This has more space, it is nice for collaboration, and your work is accessible to the rest of the group after you leave UMass. Create a directory for yourself and/or your project in <code>/work/pi_jlb_umass_edu/blanchardlab</code>. For projects that require large datasets or databases, these files will be in our <code>/project/pi_jlb_umass_edu</code> directory. However you should run all of your scripts and jobs from the work directory and have the results written to our work directory.</p>
</section>
<section id="using-software-installed-on-unity-and-through-conda" class="level2">
<h2 class="anchored" data-anchor-id="using-software-installed-on-unity-and-through-conda">Using software installed on Unity and through Conda</h2>
<p>Before you run bioinformatic tools on Unity, you first need to load the software using <a href="https://docs.unity.uri.edu/documentation/software/modules/module-usage/">Environmental Modules - Unity Docs</a>. As noted in the documentation software is often run in <a href="https://docs.unity.rc.umass.edu/documentation/jobs/sbatch/">batch mode using Slurm</a>. We often write a python script to run the sbatch file. If using the software doesn’t take much time or resources you can run it on the command line.</p>
<p>Much of the bioinformatics that we use in our lab is installed using <a href="https://docs.unity.uri.edu/documentation/software/conda/">Conda - Unity Docs</a> because often it is a suite of interacting tools that require specific versions to work correctly. When the development group releases their tools in Conda it has all of the right versions. Note that we need to start the Conda module and then activate the module specific for the bioinformatic tool.</p>
</section>
<section id="transfering-data-from-unity-to-biologys-network-attached-storage-nas" class="level2">
<h2 class="anchored" data-anchor-id="transfering-data-from-unity-to-biologys-network-attached-storage-nas">Transfering data from Unity to Biology’s Network Attached Storage (NAS)</h2>
<p>Coming soon…directions for backing up data on Unity to Biology’s Network Attached Storage (NAS).</p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>